text/0000-custom-ca-certs.md

```diff
@@ -0,0 +1,133 @@
+# Meta
+[meta]: #meta
+- Name: Custom CA Certs
+- Start Date: 2020-03-30
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+In order to allow developers to use custom CA certs during build, this RFC:
+1. Suggests an interface between stacks and the platform to allow adding ca-certs to builds and for creating and extending builders with CA certs
+2. Suggest a `pack` UX for using certs during build and for creating and extending builders with CA certs
+
+This RFC builds on some of the ideas proposed in https://github.com/buildpacks/rfcs/pull/23
+
+# Motivation
+[motivation]: #motivation
+
+Use of custom CA certs is pervasive, especially for developers in enterprise environments. These developers need to fetch assets during build from corp artifact stores, publish to registries with custom certs, and make certs available during runtime for container images. Furthermore, developers might want to more granularly apply certificates, for example leaving certs out of a runtime container that were available during build. In order to account for the possible diversity of methods for adding CA certs in various stacks, the stack itself must embed this knowledge. But we should avoid the possibility of ad-hoc methods and stack forking, by coming up with a community solution.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Specification
+Stack authors may include an `add-cert` executable to the build and/or run images of their stack:
+```
+/cnb/image/{build,run}/add-cert <cert.pem>
+```
+
+The platform will invoke this executable for each cert used during an build. This executable should take a single argument, and serve as a wrapper for adding certs to the particular stack in question. Platforms use the certs available on the build image for publishing to registries. All lifecycle phases that could do egress will have build certs.
+
+### UX
+
+##### pack
+Given the following command:
+`pack build --cert /cert1.pem --cert:build /cert2.pem --cert:run /cert3.pem`
```

Comment:
I am not sure if it makes sense to add certs to the resulting run image especially as the default option. 

The certs will obvious be needed at build time. However, it is a common practice for the platform that runs the resulting image to provide the necessary certs and rotating them when necessary.

Reply:
Thank makes sense, but it would be nice to have some mechanism for baking them in too.

Reply:
If we leave the run image out of scope for now, does that obviate the need to do rebase along with this piece of work?

Reply:
Yes 🙂